bcgov/PIMS
# PIMS-693 Update .github workflow and scripts for app-npm-dep-check

## 🎯 Summary

  
[PIMS-693: ](https://citz-imb.atlassian.net/browse/PIMS-693)


This update will allow us to parse through a list of folders and check for dependency updates in each. 


## Testing instructions

To test this change to the github action:
1. open a new tab or window and navigate to the PIMS repo: https://github.com/bcgov/pims
2. Navigate to actions tab within the PIMS repo
3. Select "App NPM Dependency Updates" from left hand menu
4. Find the drop down "Run workflow" and select it
5. Drop down and select the branch "PIMS-693" under "use workflow from"
6. select Run workflow
7. Wait until the tasks have completed and check the PIMS backlog (https://citz-imb.atlassian.net/jira/software/c/projects/PIMS/boards/26/backlog?issueLimit=100)
8. You should see 3 new parent tasks created with the following child tickets:
    - frontend Dependency Updates (DATE)
        - Update prettier...
    - react-app Dependency Updates (DATE)
        - Update prettier...
    - express-api Dependency Udates (DATE)
        - Update eslint...

Please note that these are the expected tickets as of Jan 25th @ 2:20 pm. If more tickets are created message me (Taylor) and I can check if they should be there manually. 

## 🔰 Checklist


> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.
